### ISSUE TITLE
web: Modifier should support mouse events similar to Desktop Compose

### ISSUE THREAD

**Issue by bitspittle** [JetBrains/compose-jb#1030]
In particular: https://github.com/JetBrains/compose-jb/tree/master/tutorials/Mouse_Events

If for some reason, the API intentionally diverges due to differences in how web and desktop work, that's fine -- but there needs to be a way to support onMouseEnter, onMouseLeave, and onMouseMove at the very least.

**Comment by bitspittle** [comment]
Note: I fully expect for you to diverge from how I'm doing it, but here's what I'm doing at the moment as a workaround:

https://github.com/bitspittle/kobweb/blob/main/lib/compose-ext/src/jsMain/kotlin/kobweb/compose/ui/ModifierExtensions.kt#L64

**Comment by eymar** [comment]
Perhaps I don't get the question. Let's figure it out.

Currently there is a way to add listeners that you're asking about:
[onMouseEnter](https://github.com/JetBrains/compose-jb/blob/master/web/core/src/jsMain/kotlin/org/jetbrains/compose/web/attributes/EventsListenerBuilder.kt#L47)
[onMouseLeave](https://github.com/JetBrains/compose-jb/blob/master/web/core/src/jsMain/kotlin/org/jetbrains/compose/web/attributes/EventsListenerBuilder.kt#L51)
[onMouseMove](
https://github.com/JetBrains/compose-jb/blob/master/web/core/src/jsMain/kotlin/org/jetbrains/compose/web/attributes/EventsListenerBuilder.kt#L55)

And as I see in your code snippet you're already using them.

So my question is:
What kind of mouse events support would you like to have?

**Comment by bitspittle** [comment]
My apologies for the lack of clarity, thanks for following up with those questions.

I updated the title, hopefully it's more clear now. I think mouse events should be exposed through the _Modifier_ class, beyond just `clickable` which is available right now.

FWIW, the code I linked to it in a module in my project called `compose-ext` which I'm hoping one day to _delete entirely_. I don't think it should exist - I'm just patching what I think are the current holes in the API so I can keep moving.
